qa/workunits/rbd/rbd_support_module_recovery.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+set -ex
+
+POOL=rbd2
+IMAGE_PREFIX=image
+MNT_DIR_PREFIX=mnt
+IMAGES=20
+RUN_TIME=3600
+
+ceph osd pool create ${POOL}
+rbd pool init ${POOL}
+rbd mirror pool enable ${POOL} image
+rbd mirror pool peer add ${POOL} cluster1
+
+# Create images and schedule their mirror snapshots
+for ((i=1;i<=${IMAGES};i++)); do
+    rbd create ${IMAGE_PREFIX}$i --size 1024 --pool ${POOL}
+    rbd mirror image enable ${POOL}/${IMAGE_PREFIX}$i snapshot
+    rbd mirror snapshot schedule add -p ${POOL} --image ${IMAGE_PREFIX}$i 1m
+done
+
+# Run fio workloads don images via kernel client
+for ((i=1;i<=${IMAGES};i++)); do
+    DEVS[$i]=$(sudo rbd device map ${POOL}/${IMAGE_PREFIX}$i)
+    sudo mkfs.ext4 ${DEVS[$i]}
+    MNTPATHS[$i]=`pwd`/${MNT_DIR_PREFIX}${IMAGE_PREFIX}$i
+    mkdir -p ${MNTPATHS[$i]}
+    sudo mount ${DEVS[$i]} ${MNTPATHS[$i]}
+    sudo chown $(whoami) ${MNTPATHS[$i]}
+    fio --name=fiotest --filename=${MNTPATHS[$i]}/test --rw=randrw --bs=4K \
+        --direct=1 --ioengine=libaio --size=800M --iodepth=2 \
+        --runtime=${RUN_TIME} --time_based &> /dev/null &
```

Comment:
Nice suggestion! I When I tried doing this, on a few occassions, I observed that sending either SIGTERM or SIGKILL to the specific fio background processes did not end up killing them for more than an hour so. Each fio background process results in a child and a grandchild process. I can acquire the list of all the descendants of each fio background process and kill them. Maybe it's okay to use `killall -e fio`  since more than one test won't run in a teuthology node?

Reply:
It looks like a bug in fio to me. Though knowing it does not help here.

Ideally, I would prefer if there were no assumption that the script would be running on a teuthology node only, but anyway, if `killall -e fio` works for you better then I would say just go ahead and use it. I don't think we want to complicate this.